### PR TITLE
Deadline calendar for task and routine entities

### PR DESCRIPTION
**Pull request by ihsangkcl**
Added calendar date selection feature to the ‘edit task page’, ‘edit routine page’. #315 
There are some major changes in the app after this implementation. This branch will be merged after merging these changes to the android/dev branch.

### CODE REVIEW

File: application/android/app/src/main/java/com/group12/beabee/views/entities/TaskFragmentEdit.java [hunk 1]
```diff
@@ -72,6 +88,17 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
         etTitle.setText(taskDTO.title);
         etDescription.setText(taskDTO.description);
         cbIsDone.setChecked(taskDTO.isDone);
+        tvDeadline.setText(taskDTO.deadline);
+        //Button button = (Button) findViewById(R.id.btn_pickDate);
```
Comment (yamanogluberk) [inline application/android/app/src/main/java/com/group12/beabee/views/entities/TaskFragmentEdit.java:92]: You can add onclicklisteners both ways, but again commented out code

**Review by yamanogluberk — CHANGES_REQUESTED** [review]
There are plenty of commented out codes, it is better if you can clean them up. Other than that it seems fine :)

File: application/android/app/src/main/java/com/group12/beabee/views/entities/RoutineFragmentEdit.java [hunk 2]
```diff
@@ -115,4 +131,34 @@ protected PageMode GetPageMode() {
...
+        //String currentDateString = DateFormat.getDateInstance(DateFormat.FULL).format(c.getTime());
+
+
+        // tvDeadline.setText(currentDateString);
+
+    }
+
+    @OnClick(R.id.btn_pickDate)
+    public void onClick(View view) {
+//        Calendar c = Calendar.getInstance();
```
[the remark this answers is not in the document]
Comment (ihsangkcl) [reply application/android/app/src/main/java/com/group12/beabee/views/entities/RoutineFragmentEdit.java]: I removed these lines. Thanks

File: application/android/app/src/main/java/com/group12/beabee/views/entities/TaskFragmentEdit.java (hunk 1 above)
Comment (ihsangkcl) [reply application/android/app/src/main/java/com/group12/beabee/views/entities/TaskFragmentEdit.java]: I removed these lines. Thanks

**Review by Refref18 — APPROVED** [review]

**Comment by ihsangkcl** [comment]
All deadline calendars added(task,routine,goal,subgoal,groupgoal). Period selection for routine added. Some bugs solved. I think it is ready to merge

**Review by yamanogluberk — APPROVED** [review]

**Pull request merged**